walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -18,99 +18,154 @@
 import java.util.*;
 
 public final class FindMeetingQuery {
+  /** Finds and returns a collection of non-overlapping time intervals that are
+  * each at least as long as needed for the required attendees to have a meeting
+  * of the desired length. An empty list is returned if no such times exist.
+  *
+  * @param events    - Collection of events signifying when an attendee cannot meet
+  * @param request   - Request specifying the meeting's attendees and duration
+  *                                 - Contains both required and optional attendees
+  *
+  * @return          - Collection of non-overlapping time intervals
+  */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    Collection<TimeRange> results = new ArrayList<>();
-
-    List<Event> allEvents = new ArrayList(events);
-    Collection<String> allAttendees = request.getAttendees();
-    int numAttendees = allAttendees.size();
-    long meetingTime = request.getDuration();
-
-    // Base Case: No required attendees.
-    if (numAttendees == 0) {
-        results = Arrays.asList(TimeRange.WHOLE_DAY);
-        return results;
+    List<Event> allEvents                   = new ArrayList(events);
+    Collection<String> allAttendees         = request.getAttendees();
```

Comment:
I would name this `allRequiredAttendees`.

Reply:
Done.